src/common/bustub_instance.cpp

```diff
@@ -0,0 +1,80 @@
+#include "common/bustub_instance.h"
+#include <fmt/format.h>
+#include "binder/binder.h"
+#include "catalog/table_generator.h"
+#include "recovery/log_manager.h"
+
+namespace bustub {
+
+auto BustubInstance::GetExecutionContext(Transaction *txn) -> std::unique_ptr<ExecutorContext> {
+  return std::make_unique<ExecutorContext>(txn, catalog_, buffer_pool_manager_, transaction_manager_, lock_manager_);
+}
+
+BustubInstance::BustubInstance(const std::string &db_file_name) {
+  enable_logging = false;
+
+  // storage related
+  disk_manager_ = new DiskManager(db_file_name);
+
+  // log related
+  log_manager_ = new LogManager(disk_manager_);
+
+  buffer_pool_manager_ = new BufferPoolManagerInstance(BUFFER_POOL_SIZE, disk_manager_, log_manager_);
+
+  // txn related
+  lock_manager_ = new LockManager();
+  transaction_manager_ = new TransactionManager(lock_manager_, log_manager_);
+
+  // checkpoints
+  checkpoint_manager_ = new CheckpointManager(transaction_manager_, log_manager_, buffer_pool_manager_);
+
+  // catalog
+  catalog_ = new Catalog(buffer_pool_manager_, nullptr, nullptr);
+}
+
+auto BustubInstance::ExecuteSql(const std::string &sql) -> std::vector<std::string> {
+  if (!sql.empty() && sql[0] == '\\') {
+    // internal commands
+    if (sql == "\\dt") {
+      std::string result;
+      auto table_names = catalog_->GetTableNames();
+      for (const auto &name : table_names) {
+        result += name;
+        result += "\n";
+      }
+      return {result};
+    }
+    throw Exception(fmt::format("unsupported internal command: {}", sql));
+  }
+
+  bustub::Binder binder;
+  binder.ParseAndBindQuery(sql, *catalog_);
+  std::vector<std::string> result = {};
+  for (auto &&statement : binder.statements_) {
+    result.push_back(statement->ToString());
+  }
+  return result;
+}
+
+void BustubInstance::GenerateTestTable() {
```

Comment:
We could move this to a test fixture, right?

Reply:
It is a test fixture indeed for executor tests. For this `GenerateTestTable`, it's used in the shell to predefine some tables, as we don't support create / drop table and insert for now.

Reply:
OK, could you please add a TODO?